scripts/performance/measure-performance.js

```diff
@@ -0,0 +1,186 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+// @ts-check
+const puppeteer = require('puppeteer');
+const fs = require('fs');
+const resolve = require('path').resolve;
+const path = resolve('./workspace');
+const profilesPath = './profiles/';
+
+const lcp = 'Largest Contentful Paint (LCP)';
+const performanceTag = braceText('Performance');
+
+let name = 'Measurement';
+let url = 'http://localhost:3000/#' + path;
+let folder = 'profile';
+let headless = true;
+let runs = 10;
+
+(async () => {
+    // Wait until the browser page is available
+    const browser = await puppeteer.launch({ headless: true });
+    const page = await browser.newPage();
+    await retryConnection(() => page.goto('http://localhost:3000'), 1000);
+    browser.close();
+
+    let defaultUrl = true;
+    const args = process.argv.slice(2);
+    for (let index = 0; index < args.length; index++) {
+        if (args[index].startsWith('--')) {
+            let next = args[index + 1];
+            if (!next.startsWith('--')) {
+                if (args[index] === '--name') {
+                    name = args[index + 1];
+                    index++;
+                }
+                if (args[index] === '--url') {
+                    url = args[index + 1];
+                    defaultUrl = false;
+                    index++;
+                }
+                if (args[index] === '--folder') {
+                    folder = args[index + 1];
+                    index++;
+                }
+                if (args[index] === '--headful') {
+                    headless = false;
+                    index++;
+                }
+                if (args[index] === '--runs') {
+                    runs = parseInt(args[index + 1]);
+                    index++;
+                }
+            }
+        }
+    }
+    if (defaultUrl) getOrCreateFolder(path);
+    getOrCreateFolder(profilesPath);
+    const folderPath = profilesPath + folder;
+    getOrCreateFolder(folderPath);
+    await measurePerformance(name, url, folderPath, headless, runs);
+})();
+
+async function measurePerformance(name, url, folder, headless, runs) {
+    const multipleRuns = runs > 1 ? true : false;
+    let runNr = 1;
+    let durations = [];
+    while (runs > 0) {
+        const browser = await puppeteer.launch({ headless: headless });
+        const page = await browser.newPage();
+
+        const file = folder + '/' + runNr + '.json';
+
+        await page.tracing.start({ path: file, screenshots: true });
+
+        await page.goto(url);
+        // @ts-ignore
+        await page.waitForSelector('.theia-preload', { state: 'detached' });
+        // This selector is for the problems indicator in the status bar, as this is one of the last elements to be rendered.
+        // @ts-ignore
+        await page.waitForSelector('.fa-exclamation-triangle', { state: 'visible' });
```

Comment:
Shouldn't this be `visible: true`?
```suggestion
        // @ts-ignore
        await page.waitForSelector('.fa-exclamation-triangle', { visible: true });
```